### DISCUSSION TITLE
FEATURE REQUEST: Allow disabling auto-import of custom objects when loading save

### DISCUSSION THREAD

**Discussion by player49256** [OpenRCT2/OpenRCT2#15759] [Feature requests]
**OS:** Win 10
**Version:** 0.2.0
**Commit/Build:** 0aff800

Basicly an option that lest you check/uncheck this feature, like auto-export custom objects on saves does today.

This would be useful for not cluttering up your custom objects when making scenarios for example, while still being able to load saves with tonnes of custom objects (e.g Workbenchs).

It used to be common to include scenery from expansions as custom scenery to make parks loadable for games without expansions. The result of this was alot of duplicate objects in the scenario editor. Also, loading several workbenches can net you with hundreds of custom objects you may not want when building your own scenarios.

This option would remedy this, while still having the option to import by checking the option and reload the game, or having another option in file drop-down that says "Import custom objects from savefile".

- [X] Reproducible in RCT2 (vanilla)?
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. Load save file
2. Open scenario editor
3. Check custom objects

**Dump file**
N/A

**Screenshots / Video:**
N/A

**Save game:**
N/A

**Comment by Broxzier** [comment]
I'd prefer a confirmation box over yet another option. This confirmation box would only be shown when a save contains new objects, so it won't annoy the player when loading saves they have opened before already.

**Comment by player49256** [comment]
@Broxzier I would actually agree with that.

Whenever you open a save with new objects (Not already imported). "New objects have been found within this save, do you want to import them for later use?" (Or something like that).

Maybe a help button which explains what objects are and what the impact is.

Yes / No

**Comment by Ryder17z** [comment]
@player49256 I think that it might be reasonable to consider a checkbox too;


[  ] Enable automatic import of custom objects
(legacy behavior)
